### ISSUE TITLE
Can't refresh explorer when create a new file or delete a file in a complex project

### ISSUE THREAD

**Issue by kpge** [theia-ide/theia#5919]
### Description
Can't refresh explorer when create a new file or delete a file in a complex project, but can refresh in a simple project.
in a complex project, such as theia :
![can not reresh itself](https://user-images.githubusercontent.com/35068357/62855604-4aed0e00-bd25-11e9-84be-dd0bf2016de4.gif)
but the `auto-refresh` works fine in a simple project:
![can reresh itself](https://user-images.githubusercontent.com/35068357/62855829-d23a8180-bd25-11e9-864f-e71fcbe2f518.gif)

### Reproduction Steps

**OS and Theia version:**
theia: 11badc80598254e8c78a204b7b93c2368a300fe4
ubuntu: 18.04
**Diagnostics:**

**Comment by akosyakov** [comment]
Here seems to be 2 issues:
- a bug that the navigator is not auto-refreshed
- no ability to force refresh, see also https://github.com/theia-ide/theia/issues/5885

@kpge Help is welcomed. Would you like to have a look what is going wrong?

**Comment by kpge** [comment]
ok, I will have a look at them.

**Comment by kpge** [comment]
this is duplicate with #239 and #938, 
after I increased my default inotify watches, it is resloved:
`echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p`
I think we should show a warning message in the browser to notify user to change the default config or click force refresh button.

**Comment by akosyakov** [comment]
@kpge if it is not invasive then it could be helpful. We have `Troubleshooting` section here which can be referenced as a link from the notification.

**Comment by vince-fugnitto** [comment]
@kpge is there anything else left to do with this issue after #5885 ?

**Comment by kpge** [comment]
@vince-fugnitto it can be  closed,  and  I will open a new issue to show a warning message
